### PR TITLE
Cache page metadata on first request

### PR DESCRIPTION
**Pull request by technosophos**
During testing, I realized that every page on the site was being reloaded from disk on every single page request. This PR caches page metadata so that navigation, listings, and so on can be generated from a cache without doing so much file IO

This PR adds page caching:

- A page cache is written to `config/_cache.json` that has the page metadata cached. Because `config/` is not accessible to template file loaders, this is a safe place to put a cache.
- The cache is generated on startup (once).
- The new `load_page` template function loads a page body only when needed
- Wagi can disable caching via the `-e DISABLE_CACHE=1` env var.
- The Makefile's `make serve`, which is a developer target, disables page caching for development
- Core templates have been updated to use `load_page` where necessary

In the future, we might want to make the page cache expire periodically. Right now, given the way Hippo deploys Wagi, though, it is not possible to change page content during a Wagi process's lifetime. So a static cache is very efficient.

Signed-off-by: Matt Butcher <matt.butcher@fermyon.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,3 +190,20 @@ have access to the Markdown version of the document.
...
+Bartholomew needs to make.
+
+If you are actively developing content, we suggest setting `DISABLE_CACHE=1`. By default, the `Makefile`'s `make serve`
+target disables the cache, as `make serve` is assumed to be used only for developers.
+
+### Debug Mode
+
+By default, Bartholomew does not send error messages to the browser.
+This can make it hard to debug templates and Rhai scripts.
+To send error messages to the browser, set `-e DEBUG_MODE=1`.
```
Comment (vdice) [inline README.md]: Is `DEBUG_MODE` the correct env var, or `SHOW_DEBUG`?  (Or both work?)

File: Makefile [hunk 2]
```diff
@@ -12,7 +13,8 @@ build:
 .PHONY: serve
 serve: build
 serve:
-	$(WAGI) -c ./modules.toml --log-dir ./logs -e PREVIEW_MODE=$(PREVIEW_MODE) -e SHOW_DEBUG=$(SHOW_DEBUG)
+	rm -rf config/_cache.json
```
Comment (vdice) [inline Makefile]: To enable scenarios where devs do want to use the cache (DISABLE_CACHE=0), we'd not want to run this command, right?  Maybe we could put it into a separate `clean` target and/or optionalize its execution...

Comment (technosophos) [reply Makefile]: Ah, yeah -- good call.

**Comment by radu-matei** [comment]
What if this cached templates and scripts as well?

**Comment by technosophos** [comment]
After a lot of hunting, we rooted out the performance problems, and they were not related to file loading/IO. So for now, we will leave this PR hanging and focus our optimization efforts elsewhere

**Comment by technosophos** [comment]
Closed won't fix.

**Pull request closed without merging**